scalalib/src/mill/javalib/android/AndroidModule.scala

```diff
@@ -706,4 +707,109 @@ trait AndroidModule extends JavaModule {
     PathRef(signedApk)
   }
 
+  /** ProGuard/R8 rules configuration files (user-provided and generated) */
+  def proguardReleaseConfigs: T[Seq[PathRef]] = {
+    Seq(
+      PathRef(moduleDir / "proguard-rules.pro"),
+      PathRef(androidModuleGeneratedSourcesFunc().androidReleaseDex.path / "proguard-rules.pro"),
+      PathRef(moduleDir / "test-proguard-rules.pro")
+    )
+  }
+
+  def proguardDebugConfigs: T[Seq[PathRef]] = {
+    Seq(
+      PathRef(moduleDir / "proguard-rules.pro"),
+      PathRef(androidModuleGeneratedSourcesFunc().androidDebugDex.path / "proguard-rules.pro"),
+      PathRef(moduleDir / "test-proguard-rules.pro")
+    )
+  }
+
+  // Additional library classes (if any) that should be treated as provided
+  def libraryClasses: T[Seq[PathRef]] = T {
+    Seq(
+      androidSdkModule().androidJarPath(),
+      androidSdkModule().r8LibPath(),
+      androidSdkModule().r8LibClasspath(),
+      androidSdkModule().androidCoreModulesPath(),
+      androidSdkModule().androidOptionalApacheHttpLegacy(),
+      androidSdkModule().androidOptionalCar(),
+      androidSdkModule().androidOptionalTestMock(),
+      androidSdkModule().androidOptionalTestBase(),
+      androidSdkModule().androidOptionalTestRunner()
+    )
+  }
+
+  def enableDesugaring: T[Boolean] = Task {
+    true
+  }
+
+  // Optional multi-dex rules or main-dex list files for legacy multidex support
+  def mainDexRules: T[Option[PathRef]] = T {
+    Some(PathRef(androidResources()._1.path / "main-dex-rules.pro"))
+  }
+
+  def mainDexList: T[Option[PathRef]] = T {
```

Comment:
Task is named "List" but type can only contain one element. If for compat with other build tools this name can't be changed, add a good comment.